src/libraries/Microsoft.PowerFx.Interpreter/Functions/Mutation/CollectFunction.cs

```diff
@@ -298,11 +298,12 @@ public virtual async Task<FormulaValue> InvokeAsync(FormulaValue[] args, Cancell
             }
 
             var tableValue = arg0 as TableValue;
-            var recordValue = arg1 as RecordValue;
-            var recordValueCopy = FormulaValue.NewRecordFromFields(recordValue.Fields);
+            var recordValue = arg1 as RecordValue;
+
+            var recordValueToAppend = (RecordValue)recordValue.MaybeShallowCopy();
```

Comment:
>RecordValue [](http://example.com/codeflow?start=39&length=11)

nit: already a cast at L301

Reply:
Yes but `MaybeShallowCopy` returns a FormulaValue. That's why this extra cast.